pkg/api/instances.go

```diff
@@ -344,11 +390,19 @@ func (api *API) updateInstanceStatus(instanceID, appID string, newStatus int) er
 	if err != nil {
 		return err
 	}
+	return api.updateInstanceObjStatus(instance, newStatus)
+}
+
+func (api *API) updateInstanceData(instance *Instance, data *map[string]interface{}) error {
```

Comment:
There is no need to make `data` a pointer to map. Map types are pointer-like themselves, so passing them "by value" is cheap. So:
```suggestion
func (api *API) updateInstanceData(instance *Instance, data map[string]interface{}) error {
```

Reply:
This change gets me nervous, but you're coming from C++ too, so I will accept it.

Reply:
Oh yeah, from that angle I can understand why it makes you nervous. Coming from that background, maps can have a bit weird semantics, see https://play.golang.org/p/RMG9kLd4d2L.